### DISCUSSION TITLE
jupyter

### DISCUSSION THREAD

**Discussion by Paul8043** [bernhard-42/vscode-ocp-cad-viewer#26] [Q&A]
Is there a working example which shows how jupyter should be used in the ocp-context?

**Comment by jdegenstein** [comment]
Are you speaking of the interactive console?

**Reply by Paul8043** [reply]
I know jupyter-notebooks more in the sense of  a mathematica-notebook. It contains text together with formulas/programs and graphical output, all in a single document. Such document offers a limited interactivity in the sense you can change a few parameter or code and you can see immediately the impact of your modification.
This could be an appropriate tool for writing tutorials for Build123d.
 
An interactive console, for its own, might be nice to have, but I am not a friend of REPL (read-eval-print loop). I don't think this makes much sense for an interpreted language like python.

**Reply by bernhard-42** [reply]
There is [Jupyter-CadQuery](https://github.com/bernhard-42/jupyter-cadquery) - which I will update to the latest ocp-tessellate, three-cad-viewer and behavior of OCP CAD Viewer when I find time ...
It should work with build123d though

**Comment by Paul8043** [comment]
Good to hear, not needed immediately. Just want to know what might come.
